src/device-config.coffee

```diff
@@ -259,7 +259,16 @@ module.exports = class DeviceConfig
 		systemd.serviceActiveState(logToDisplayServiceName)
 		.then (activeState) ->
 			return activeState not in [ 'inactive', 'deactivating' ]
-		.catchReturn(UnitNotLoadedError, null)
+		.catch (e) ->
+			# Depending on whether the error is a 'unit not loaded' error, or
+			# otherwise, we need to return the correct value
+			if e[0] is "Unit #{logToDisplayServiceName}.service not loaded."
```

Comment:
Why did you stop using the `UnitNotLoadedError` predicate?  It looks like this is the only place using it

Reply:
Because it looks like that was exclusively from the go supervisor. Either way, the error that now comes through, `typeof` reports object, and none of the standard error fields (message, stack etc) are present. From what I can tell, this comes from the node dbus library, so we can fix it properly coming up, but this is quite a serious bug, so I'd like to fix it as-is for now.

Reply:
Looks like it comes from here: https://github.com/sidorares/dbus-native/blob/7ddc7ebd0b6b988177733a8ac4b2f0aeff70cc73/lib/bus.js#L105 I'd definitely rather not rely on parsing output to detect errors though, I just can't think of a better method just now.